apiserver/facades/client/resources/repository.go

```diff
@@ -63,7 +62,20 @@ func (ch *charmHubClient) ResolveResources(uploadedResources []charmresource.Res
 		return nil, errors.Trace(err)
 	}
 	for _, res := range uploadedResources {
-		chResources[res.Name] = res
+		if res.Origin == charmresource.OriginUpload {
+			chResources[res.Name] = res
+			continue
+		}
+		// TODO (hml) 07-dec-2020
+		// Where to find the metadata from the charmhub api? For now,
```

Comment:
We found how to get resource.Meta, however it requires a change to the charmhub api, which is not currently available.  This is a work around to facilitate development and testing of this functionality.